kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiBatchSessionImpl.scala

```diff
@@ -113,6 +113,11 @@ class KyuubiBatchSessionImpl(
       ms.incCount(MetricRegistry.name(CONN_OPEN, user))
     }
 
+    // have to add batchJobSubmissionOp into session opHandleSet here, to update the batch metadata
+    // on openSession failure(may be open session for recovery), otherwise, the batch metadata state
+    // might lose control.
+    opHandleSet.add(batchJobSubmissionOp.getHandle)
```

Comment:
does `runOperation(batchJobSubmissionOp)` not work ?

Reply:
it might fail before `runOperation(batchJobSubmissionOp)`

Reply:
let me try to reorder them

Reply:
it should be fine now.